cmd/kubeadm/app/phases/controlplane/users_linux.go

```diff
@@ -0,0 +1,260 @@
+// +build linux
+
+package controlplane
+
+import (
+	"bufio"
+	"bytes"
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"strconv"
+	"strings"
+
+	kubeadmconstants "k8s.io/kubernetes/cmd/kubeadm/app/constants"
+	"k8s.io/utils/exec"
+)
+
+var ()
+
+var UpdateFileOwnershipFn = func(filepath string, uid int64, gid int64, permissions uint32) error {
```

Comment:
This needs a comment and warning that this should not be called directly and is only for testing to override the function behavior for tests.

Also I hate that I have to do this and am open to suggestions on doing it another way if possible.

Reply:
the function that is being unit tested can accept a function pointer.
during regular execution it can accept UpdateFileOwnership, else it can accepted one for testing.

ideally this here should be `func UpdateFileOwnership(...)`